src/main/java/com/redhat/devtools/lsp4ij/commands/CommandExecutor.java

```diff
@@ -164,15 +162,64 @@ private static AnAction createIDEACoreCommand(Command command) {
         return ActionManager.getInstance().getAction(commandId);
     }
 
-    private static DataContext createDataContext(Project project, Command command, URI documentUri) {
-
+    private static DataContext createDataContext(URI documentUri, Command command, AnAction action, Component source, Project project) {
         SimpleDataContext.Builder contextBuilder = SimpleDataContext.builder();
-        contextBuilder.add(CommonDataKeys.PROJECT, project)
-                    .add(LSP_COMMAND, command)
-                    .add(LSP_COMMAND_DOCUMENT_URI, documentUri);
+        if (source != null) {
+            contextBuilder.setParent(DataManager.getInstance().getDataContext(source));
+        }
+        updateCommandArguments(command, action.getClass().getClassLoader());
+        contextBuilder
+                .add(CommonDataKeys.PROJECT, project)
+                .add(LSP_COMMAND, command);
+        if (documentUri != null) {
+            contextBuilder.add(LSP_COMMAND_DOCUMENT_URI, documentUri);
+        }
         return contextBuilder.build();
     }
 
+    private static void updateCommandArguments(Command command, ClassLoader classLoader) {
+        List<Object> arguments = command.getArguments();
+        if (arguments == null || arguments.isEmpty()) {
+            return;
+        }
+        for (int i = 0; i < arguments.size(); i++) {
+            Object arg = arguments.get(i);
+            if (arg instanceof JsonElement) {
+                JsonElement elt = (JsonElement) arg;
+                if (elt.getClass().getClassLoader() != classLoader) {
```

Comment:
if (classloader == elt.getClass().getClassLoader() || !(classLoader instanceof PluginClassLoader) {
    return; // classloader won't change with the next argument
}

Reply:
fixed